### PR TITLE
finish some of any and do some code formatting

### PR DESCRIPTION
**Pull request by mikecovlee**
Well, our any now can run fast with small object optimize

### CODE REVIEW

File: shared/include/covscript/shared/any.hpp [hunk 1]
```diff
@@ -1,102 +1,213 @@
...
+	};
+	/*
+	    实现小对象优化
+	    减少内存分配瓶颈
+	*/
+	enum class stor_status {
+		null, ptr, data
+	};
+
+	struct stor_union {
```
Comment (ice1000) [inline shared/include/covscript/shared/any.hpp:108]: What does "stor" stand for?

**Review by ice1000 — COMMENTED** [review]
As a formatting PR, why both `a = b` and `a=b` are kept? Do we add spaces around assignment operators or don't we?

File: shared/include/covscript/shared/any.hpp (hunk 1 above)
Comment (imkiva) [reply shared/include/covscript/shared/any.hpp:108]: Storage I think.

File: shared/include/covscript/shared/any.hpp [hunk 2]
```diff
@@ -1,102 +1,213 @@
...
+	class type_support;
+	namespace runtime {
+		class any;
+	}
 }
 
-class cs::impl::any final {
+class cs::runtime::any final {
 public:
-    using typeid_t=std::type_index;
```
Comment (imkiva) [inline shared/include/covscript/shared/any.hpp:14]: Please just add some spaces around `=`

File: shared/include/covscript/shared/exception.hpp [hunk 3]
```diff
@@ -28,17 +28,17 @@ namespace cs {
...
 
-    namespace event{
-        extern utility::event_type exception_raised;
-    }
-
-    template<typename T, typename...ArgsT>
-    void throw_ex(ArgsT&&...args)
-    {
-        T exception(std::forward<ArgsT>(args)...);
-        std::exception &stdexcept=exception;
```
Comment (imkiva) [inline shared/include/covscript/shared/exception.hpp]: Spaces here pls :)

File: shared/include/covscript/shared/exception.hpp [hunk 4]
```diff
@@ -28,17 +28,17 @@ namespace cs {
...
-    }
+	namespace event {
+		utility::event_type exception_raised;
+	}
+
+	template<typename T, typename...ArgsT>
+	void throw_ex(ArgsT&&...args)
+	{
+		T exception(std::forward<ArgsT>(args)...);
+		std::exception &stdexcept=exception;
```
Comment (imkiva) [inline shared/include/covscript/shared/exception.hpp]: And here....

File: shared/src/test_any.cpp [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#include <covscript/shared/any.hpp>
+#include <iostream>
+#include <string>
+
+int main()
+{
+	cs::runtime::any a, b(10), c(std::string("Hello"));
+	cs::runtime::any d(a), e(b), f(c);
+	std::cout<<c.get<std::string>()<<std::endl;
```
Comment (imkiva) [inline shared/src/test_any.cpp]: PLS add some spaces around `<<`

**Review by imkiva — CHANGES_REQUESTED** [review]
Spaces. Spaces. Spaces.....
Please just keep that in mind.....
I have been saying that for 3 thousand times.........

**Comment by ice1000** [comment]
Sticky spaces

**Comment by imkiva** [comment]
Sticky sapces

**Review by imkiva — COMMENTED** [review]
Accepted.

**Comment by ice1000** [comment]
You should say "approved"

**Comment by imkiva** [comment]
Well, "Approved" next time.

**Pull request merged**